.circleci/verbatim-sources/linux-binary-build-defaults.yml

```diff
@@ -58,6 +60,8 @@ binary_linux_test: &binary_linux_test
       at: ~/workspace
   - attach_workspace:
       at: /home/circleci/project
+  - run:
+      <<: *should_run_job
```

Comment:
I don't think you need this or the one on upload. upload jobs always follow test jobs always follow build jobs

Reply:
Actually I do, because `circleci step halt` only halts the step in the workflow; CircleCI will attempt to run the next step (test, upload). So need to cancel in all cases :)

Reply:
:/